packages/engine/src/scene/components/PointLightComponent.ts

```diff
@@ -11,6 +11,19 @@ import { ObjectLayers } from '../constants/ObjectLayers'
 import { setObjectLayers } from '../functions/setObjectLayers'
 import { addObjectToGroup, removeObjectFromGroup } from './GroupComponent'
 
+export type PointLightComponentType = {
```

Comment:
We can use ComponentType<typeof PointLightComponent> for these types

Reply:
Why not put the castShadow property with all of the other shadow properties that are in the light components?

Reply:
Also, the lights need different default values for castShadow than meshes and whatnot (lights false, meshes true)

Reply:
> Why not put the castShadow property with all of the other shadow properties that are in the light components?

That's a good point, in this case we should remove remove castShadow from ShadowComponent and turn it into a tag component

Reply:
@HexaField what about receiving shadows? You want a tag component for casting and receiving each?

Reply:
Sorry, yes, I mean put castShadow on light components, and have receiveShadow be set only via the presence of a tag ShadowComponent